### DISCUSSION TITLE
Regex help for Home Assistant `build.json` file

### DISCUSSION THREAD

**Discussion by loganmarchione** [renovatebot/renovate#18380] [Request Help] [answered]
Home Assistant add-ons use a `build.yaml` or `build.json` file when building their `Dockerfile`.

In my case, I'm using a [build.json](https://github.com/loganmarchione/hassos-addons/blob/main/prometheus_node_exporter/build.json) file.

```
{
  "build_from": {
    "amd64": "ghcr.io/hassio-addons/base/amd64:12.2.0",
    "aarch64": "ghcr.io/hassio-addons/base/aarch64:12.2.0",
    "armv7": "ghcr.io/hassio-addons/base/armv7:12.2.0"
  }
}
```

I'm struggling with the regex match for this. Can anyone assist? Using regex101.com, I'm currently using this.

```
\s*ghcr.io\/hassio-addons\/base\/.*:(?<major>\d+).(?<minor>\d+).(?<patch>\d+)
```

Also, I'm confused if this needs to be in `regexManagers` or in `packageRules`. I think `packageRules`, right?

```
  "packageRules": [
    {
      "matchDatasources": ["docker"],
      "matchPackagePatterns": ["ghcr.io/hassio-addons/base"],
      "versioning": "regex:\s*ghcr.io\/hassio-addons\/base\/.*:(?<major>\d+).(?<minor>\d+).(?<patch>\d+)"
    },
   ]
```

**Comment by RahulGautamSingh** [comment, marked as answer]
Hey @loganmarchione ,
You will need to add `regexManagers` to your config and use `fileMatch` so that the bot can know which file to extract from. 
You can use `matchStrings` which are regexes that help the bot match the strings which contain package info liike depName and version value. 

I have attached a working config below. Here is the [repo ](https://github.com/RahulGautamSingh-testing/hassos-addons/pulls)I tested it on.
```
"regexManagers": [
    {
      "fileMatch": ["(^|/|\\.)build.json"],
      "matchStringsStrategy": "any",
      "matchStrings": [
         "\"amd64\": \"(?<depName>.*?):(?<currentValue>.*?)\"",
          "\"aarch64\": \"(?<depName>.*?):(?<currentValue>.*?)\"",
          "\"armv7\": \"(?<depName>.*?):(?<currentValue>.*?)\""
      ],
      "datasourceTemplate": "docker"
    }
  ]
  ```

**Reply by loganmarchione** [reply]
Wow thank you, working perfectly! 🙏 I was way off! Plus this serves as a good example for some other projects I have, thanks!
